### PR TITLE
chore: Make codegen independent from protoc

### PR DESCRIPTION
**Pull request by tottoto**
Makes `codegen` independent from `protoc` by using `protox`. This eliminates the need to match `protoc` versions in development, and separates the public crates from `protoc`.

### CODE REVIEW

File: codegen/src/main.rs [hunk 1]
```diff
@@ -105,3 +114,10 @@ fn codegen(
         std::fs::copy(&path, &to).unwrap();
     }
 }
+
+fn write_fds(fds: &FileDescriptorSet, path: &Path) {
+    let mut writer = BufWriter::new(std::fs::File::create(path).unwrap());
```
Comment (djc) [inline codegen/src/main.rs:119]: Nit: I'd suggest importing `File` instead.

**Review by djc — APPROVED** [review]
Nice!

File: codegen/src/main.rs (hunk 1 above)
Comment (tottoto) [reply codegen/src/main.rs]: Applied the suggestion.

Comment (tottoto) [reply codegen/src/main.rs]: Applied the suggestion.

**Pull request merged**